src/utils/alimentaide/onStructureAdded.ts

```diff
@@ -117,6 +139,9 @@ export const linkStructureAppsToDemo = async (rerun = false) => {
     application: { $in: structureApps.map((a) => a._id) },
   });
 
+  // Add the context filter from the base app to existing structure apps
+  addContextFilterToStructureApps(structureApps);
+
```

Comment:
It doesn't make sense to add this function inside onStructureAdded, as it is run every time a new structure is added and this is a "migration" of some sorts